File: solutions/ruby/collatz-conjecture/1/collatz_conjecture.rb
class CollatzConjecture
  def self.steps(number, steps = 0)
    raise ArgumentError if number <= 0

    return steps if number == 1

    steps += 1
    if number.even?
      steps(number / 2, steps)
    else
      steps(number * 3 + 1, steps)
    end
  end
end

File: solutions/c/collatz-conjecture/1/collatz_conjecture.c
#include "collatz_conjecture.h"

int steps(int start){
  int number = start;
  int steps = 0;

  if (number <= 0) { return -1; }

  while (number > 1) {
    number = number % 2 == 0 ? number / 2 : number * 3 + 1;
    steps++;
  }

  return steps;
}
